lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,217 @@ def load_team_membership_csv(course, response):
         "Team membership CSV download is not yet implemented."
     )
     response.write(not_implemented_message + "\n")
+
+
+class TeamMembershipImportManager(object):
+    """
+    A manager class that is responsible the import process of csv file including validation and creation of
+    team_courseteam and teams_courseteammembership objects.
+    """
+
+    def __init__(self, course):
+        # the list of validation errors
+        self.validation_errors = []
+        self.teamset_names = []
+        # this is a dictionary of dictionaries that ensures that a student can belong to
+        # one and only one team in a teamset
+        self.user_ids_by_teamset_index = {}
+        # dictionary that matches column index to a teamset name. Used when creating teams to get the right teamset
+        self.teamset_name_by_index = {}
+        # the currently selected user
+        self.user = ''
+        self.number_of_record_added = 0
+        # stores the course module that will be used to get course metadata
+        self.course_module = ''
+        # stores the course for which we are populating teams
+        self.course = course
+        self.max_erros = 0
+        self.max_erros_found = False
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return len(self.validation_errors) == 0
+
+    def set_team_membership_from_csv(self, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+
+        Arguments:
+            course (CourseDescriptor): Course module for which team membership needs to be set.
+        """
+        self.course_module = modulestore().get_course(self.course.id)
+        all_rows = [row for row in csv.reader(input_file.read().decode('utf-8').splitlines())]
+        header_row = all_rows[0]
+        if self.validate_teamsets(header_row):
+            # process student rows:
+            for i in range(1, len(all_rows)):
```

Comment:
You can pluck the header out of `all_rows` a few lines earlier and use list slicing to turn this into `for row in all_rows:`